src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -557,5 +570,6 @@ public boolean equals(Object other) {
                     && getAmenitiesAdded().equals(e.getAmenitiesAdded())
                     && getAmenitiesRemoved().equals(e.getAmenitiesRemoved());
         }
+
```

Comment:
I think the equals method should include checking the favourite attribute too, but this may affect the test methods.